Shared/Samples/Edit feature attachments/EditFeatureAttachmentsView.Model.swift

```diff
@@ -0,0 +1,145 @@
+                scale: 100_000_000
+            )
+            return map
+        }()
+        
+        /// The currently selected map feature.
+        private var selectedFeature: ArcGISFeature?
+        
+        /// The placement of the callout on the map.
+        var calloutPlacement: CalloutPlacement?
+        
+        /// Holds the attachments of the currently selected feature.
+        @Published var attachments: [Attachment] = []
+        
+        /// The text shown on the callout.
+        @Published var calloutText: String = ""
+        
+        /// The text shown on the callout.
+        @Published var calloutDetailText: String = ""
+        
+        /// A Boolean value that indicates whether the callout placement should be offset for the map magnifier.
+        @Published var calloutShouldOffset = false
+        
+        /// The `FeatureLayer` populated with data by the feature table using the remote service url. 
+        var featureLayer: FeatureLayer = {
+            let featureTable = ServiceFeatureTable(url: .featureServiceURL)
+            var featureLayer = FeatureLayer(featureTable: featureTable)
+            return featureLayer
+        }()
+        
+        init() {
+            map.addOperationalLayer(featureLayer)
+        }
+        
+        /// Updates the location of the callout placement to a given screen point.
+        /// - Parameter location: The screen point at which to place the callout.
+        func updateCalloutPlacement(to location: Point) {
+            // Create an offset to offset the callout if needed, e.g. the magnifier is showing.
+            let offset = calloutShouldOffset ? CGPoint(x: 0, y: -70) : .zero
+            calloutPlacement = .location(location, offset: offset)
+        }
+        
+        /// Selects the feature that is tapped.
+        /// - Parameter feature: The feature to set as the selected feature.
+        func setSelectedFeature(for feature: ArcGISFeature) async throws {
+            selectedFeature = feature
+            featureLayer.selectFeature(selectedFeature!)
+            try await fetchAttachmentsAndUpdateFeature()
+        }
+        
+        /// Updates the callout text for the selected feature.
+        private func updateCalloutDetailsForSelectedFeature() {
+            if let selectedFeature = selectedFeature {
+                let title = selectedFeature.attributes["typdamage"] as? String
+                calloutText = title ?? "Callout"
+                calloutDetailText = "Number of attachments: \(attachments.count)"
+            }
+        }
+        
+        /// Adds a new attachment with the given parameters and synch this change with the server.
+        /// - Parameters:
+        ///   - named: The attachment name.
+        ///   - type: The attachments data type.
+        ///   - dataElement: The attachment data.
+        func addAttachment(named: String, type: String, dataElement: Data) async throws {
```

Comment:
```suggestion
        ///   - name: The attachment name.
        ///   - type: The attachments data type.
        ///   - dataElement: The attachment data.
        func addAttachment(named name: String, type: String, dataElement: Data) async throws {
```
`named` should be the argument label here not the parameter name.